### DISCUSSION TITLE
Can we deal with 2d or 3D here with CMethods

### DISCUSSION THREAD

**Discussion by muhshif** [btschwertfeger/python-cmethods#23] [Q&A]
I could see the CMethods are mostly applicable to 1-Dimensional data. But, if in case we want to plot a spatial distribution (for which we need at least 2-Dimensional data or 3-Dimensional data including latitude and longitude) of bias corrected temperature or etc. over a particular region (say Germany or India) what should we do?
I saw that there you mentioned we can use a particular command "cmethods.CMethods.adjust_3d()". But, in that cast I did not understand whether we will be able to plot the temperature or any other meteorological variable spatially over a particular region.

**Comment by btschwertfeger** [comment]
The bias correction methods implemented in pyhton-cmethods are designed to adjust a single time series for a grid cell, i.e., perform 1-dimensional adjustment along the time dimension. However, there are currently no methods implemented that perform the adjustment of a time series for a grid cell based on the surrounding cells, i.e., perform 2- or 3-dimensional adjustment for a single cell.

The method "adjust_3d" offers the possibility to adjust 3-dimensional data sets. Within this method, the correction is performed for each time series iterated over all spatial indices. It is assumed that the data sets have the same spatial resolution and the dimensions "time", "lat", and "lon" are named as such.

While there is no method for adjusting 2-dimensional data sets, users can always implement such adjustments themselves. For example, assuming a 2-dimensional data set with a resolution of 10 x 10950 (lat x time), one could iterate over the 10 latitudes and execute any bias correction method for each of these time series. This could be achieved by constructing a loop around a command similar to this: https://github.com/btschwertfeger/python-cmethods/blob/v1.0.0/cmethods/__init__.py#L436-L441 and store the adjusted time series in a new data set.
